kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -99,6 +137,21 @@ private[spark] class KubernetesClusterSchedulerBackend(
     // create a single k8s executor pod.
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
+
+    val submitArgs = mutable.ArrayBuffer.empty[String]
+
+    if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
+      submitArgs ++= Vector(
+        "dynamic-executors")
```

Comment:
And before we go too far we should ask perhaps @rxin or someone familiar with shuffle if they see long term problem with this. 
Where is the code that just launchs shuffle service? I don't see it in the PR